### PR TITLE
Fix flaky testWriteDifferentSizes

### PR DESCRIPTION
**Pull request by linzebing**
## Description

https://github.com/trinodb/trino/pull/11607 we assert that query produces three files with sizes in different ranges. However it turned out that it's not deterministic, as row distribution between writers is not deterministic.

This PR addresses the flakiness by writing only one large row that exercises different code paths of S3 streaming upload.

> Is this change a fix, improvement, new feature, refactoring, or other?

Increase test coverage

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

trino-hive

> How would you describe this change to a non-technical end user or system administrator?

## Related issues, pull requests, and links

Address https://github.com/trinodb/trino/issues/11659

## Documentation

(x) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveOnDataLake.java [hunk 1]
```diff
@@ -402,4 +396,22 @@ protected void copyTpchNationToTable(String testTable)
     {
         computeActual(format("INSERT INTO " + testTable + " SELECT name, comment, nationkey, regionkey FROM tpch.tiny.nation"));
     }
+
+    protected void assertFileSizeRange(String testTable, int scaleFactorInThousands, long fileSizeRangeStart, long fileSizeRangeEnd)
```
Comment (losipiuk) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveOnDataLake.java:400]: nit: private?

also maybe rename it to `testWriteWithFileSize` or sth in the line (the method is more than just assertion)

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveOnDataLake.java [hunk 2]
```diff
@@ -263,27 +265,19 @@ public void testWriteLargeFiles()
...
-                        .row(1L)
-                        .row(2L)
-                        .build());
-        query(format("SELECT SUM(LENGTH(col1)) FROM %s", testTable))
-                .assertThat()
-                .skippingTypesCheck()
-                .containsAll(resultBuilder(getSession())
-                        .row(500L * 70 * 1000)
-                        .build());
+        assertFileSizeRange(testTable, 150, partSizeInBytes * 2 + 1, partSizeInBytes * 3);
```
Comment (losipiuk) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveOnDataLake.java:279]: How are we sure that we will not get only small files if we end up having many writer tasks?
I am probably missing sth.

Comment (linzebing) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveOnDataLake.java]: My understanding even if we have multiple writer tasks, the distribution is still on a row-basis. Given that we write only one row, then there will be only one file.

Comment (losipiuk) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveOnDataLake.java]: Oh yeah - sure :) Makes total sense.

**Pull request merged**